### PR TITLE
Square Access APIs added

### LINKED ISSUE #1184: Square

### PR DESCRIPTION
**Pull request by Zeeshan-Ethyca**
# Purpose
This PR aims to add a Square connector Customer, Order access endpoints.

# Changes
- Added config for Square Customer Access Endpoints
- Added config for Square Order Endpoints
- Added related dataset and configuration
- Added associated tests and fixtures to validate the above functionality

# Checklist
- [x] Update [`CHANGELOG.md`](https://github.com/ethyca/fidesops/blob/main/CHANGELOG.md) file
  - [x] Merge in main so the most recent `CHANGELOG.md` file is being appended to
  - [x] Add description within the `Unreleased` section in an appropriate category. Add a new category from the list at the top of the file if the needed one isn't already there.
  - [x] Add a link to this PR at the end of the description with the PR number as the text. example: [#1](https://github.com/ethyca/fidesops/pull/1)
- [ ] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [x] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
- [x] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [x] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes  #1184

### CODE REVIEW

File: tests/ops/integration_tests/saas/test_square_task.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        graph,
+        [square_connection_config],
+        {"email": square_identity_email},
+        db,
+    )
+
+    assert_rows_match(
+        v[f"{dataset_name}:customer"],
+        min_size=1,
+        keys=["id", "email_address"],
```
Comment (galvana) [inline tests/ops/integration_tests/saas/test_square_task.py:56]: Can you add the other fields for these three `assert_rows_match` calls?

**Review by galvana — CHANGES_REQUESTED** [review]
This is a good first pass! I made some minor formatting changes and left one comment. Other than that just make sure the dataset is up to date and that the PII fields are flagged with the appropriate data category.

File: tests/ops/integration_tests/saas/test_square_task.py (hunk 1 above)
Comment (Zeeshan-Ethyca) [reply tests/ops/integration_tests/saas/test_square_task.py]: added

File: data/saas/dataset/square_dataset.yml [hunk 2]
```diff
@@ -0,0 +1,341 @@
+dataset:
+- fides_key: square_instance
+  name: Square Dataset
+  description: A sample dataset representing the Square connector for Fidesops
+  collections:
```
Comment (galvana) [inline data/saas/dataset/square_dataset.yml:5]: Make sure you update the `data_categories` with the correct values, possible values (the privacy_keys) can be found here https://github.com/ethyca/fideslang/blob/main/data_files/data_categories.json

**Review by galvana — CHANGES_REQUESTED** [review]
The dataset looks more complete in terms of fields but it still needs the correct values for the data_categories, I left a comment on the dataset with some guidance.

**Pull request closed without merging**